src/server/snapshot.cc

```diff
@@ -251,6 +207,49 @@ void SliceSnapshot::IterateBucketsFb(const Cancellation* cll, bool send_full_syn
           << stats_.loop_serialized << "/" << stats_.side_saved << "/" << stats_.savecb_calls;
 }
 
+void SliceSnapshot::SwitchIncrementalFb(Context* cntx, LSN lsn) {
+  auto* journal = db_slice_->shard_owner()->journal();
+  DCHECK(journal);
+  DCHECK_LE(lsn, journal->GetLsn()) << "The replica tried to sync from the future.";
+
+  VLOG(1) << "Starting incremental snapshot from lsn=" << lsn;
+
+  // The replica sends the LSN of the next entry is wants to receive.
+  while (!cntx->IsCancelled() && journal->IsLSNInBuffer(lsn)) {
+    serializer_->WriteJournalEntry(journal->GetEntry(lsn));
+    PushSerializedToChannel(false);
+    lsn++;
+  }
```

Comment:
could you do it in FOR cycle, it will be more readable

Reply:
I just moved the code and I deliberately did not want to modify it in this PR.